src/requirements_txt.rs

```diff
@@ -1,119 +1,603 @@
+        )?;
+        eat_trailing_line(s, requirements_txt.as_ref())?;
+        RequirementsTxtStatement::Constraint {
+            filename: constraints_file.to_string(),
+            location,
+        }
+    } else if s.eat_if("-e") {
+        let (requirement, hashes) = parse_requirement_and_hashes(s, &content, &requirements_txt)?;
+        eat_trailing_line(s, requirements_txt.as_ref())?;
+        RequirementsTxtStatement::RequirementEntry(RequirementEntry {
+            requirement,
+            hashes,
+            editable: true,
+        })
+    } else if s.at(char::is_ascii_alphanumeric) {
+        let (requirement, hashes) = parse_requirement_and_hashes(s, &content, &requirements_txt)?;
+        eat_trailing_line(s, requirements_txt.as_ref())?;
+        RequirementsTxtStatement::RequirementEntry(RequirementEntry {
+            requirement,
+            hashes,
+            editable: false,
+        })
+    } else if let Some(char) = s.peek() {
+        return Err(RequirementsTxtError::Parser {
+            message: format!(
+                "Unexpected '{}', expected '-c', '-e', '-r' or the start of a requirement",
+                char
+            ),
+            file: requirements_txt.as_ref().to_path_buf(),
+            location: s.cursor(),
+        });
+    } else {
+        // EOF
+        return Ok(None);
+    }))
+}
+
+/// Eat whitespace and ignore newlines escaped with a backslash
+fn eat_wrappable_whitespace<'a>(s: &mut Scanner<'a>) -> &'a str {
+    let start = s.cursor();
+    s.eat_while(|c: char| c == ' ' || c == '\t');
+    // Allow multiple escaped line breaks
+    while s.eat_if("\\\n") || s.eat_if("\\\r\n") {
+        s.eat_while(|c: char| c == ' ' || c == '\t');
+    }
+    s.from(start)
+}
+
+/// Eats the end of line or a potential trailing comma
+fn eat_trailing_line(
+    s: &mut Scanner,
+    requirements_txt: impl AsRef<Path>,
+) -> Result<(), RequirementsTxtError> {
+    if s.eat_if("#") {
+        s.eat_while(|c| c != '\r' && c != '\n');
+    }
+
+    if s.eat_if('\r') {
+        if !s.eat_if('\n') {
+            Err(RequirementsTxtError::Parser {
+                message: "Expected \\n after \\n, found {}".to_string(),
+                file: requirements_txt.as_ref().to_path_buf(),
+                location: s.cursor(),
+            })
         } else {
-            Some("*".to_string())
-        };
-
-        let dep = poetry_toml::Dependency::Expanded {
-            version,
-            optional: Some(false),
-            extras: Some(requirement.extras.clone()),
-            git,
-            branch: requirement.revision.clone(),
-        };
-
-        let name = requirement.name.with_context(|| {
-            format!(
-                "requirement needs a name in '{}' of {}",
-                requirement.line,
-                requirements_txt.display()
-            )
-        })?;
-        poetry_requirements.insert(name, dep);
+            Ok(())
+        }
+    } else if s.eat_if('\n') || s.done() {
```

Comment:
Isn't the `\n` already handled on line 279?

```suggestion
    } else if s.done() {
```